pkg/apiutil/attribute.go

```diff
@@ -274,6 +274,28 @@ func UnmarshalAttribute(an *apb.Any) (bgp.PathAttributeInterface, error) {
 		return bgp.NewPathAttributeLargeCommunities(communities), nil
 	case *api.PrefixSID:
 		return UnmarshalPrefixSID(a)
+	case *api.LsAttribute:
+		lsAttr, err := UnmarshalLsAttribute(a)
+		tlvs := bgp.NewLsAttributeTLVs(lsAttr)
+		var length uint16
+		for _, tlv := range tlvs {
+			length += uint16(tlv.Len())
+		}
+		t := bgp.BGP_ATTR_TYPE_LS
+		pathAttributeLs := &bgp.PathAttributeLs{
+			PathAttribute: bgp.PathAttribute{
+				Flags:  bgp.PathAttrFlags[t],
+				Type:   t,
+				Length: length,
+			},
+			TLVs: tlvs,
+		}
+		if err != nil {
+			return nil, fmt.Errorf("failed to unmarshal BGP-LS Attribute: %s", err)
```

Comment:
why isn't this err checking placed right after _UnmarshalLsAttribute()_(line 278)?

Reply:
sorry, corrected in https://github.com/osrg/gobgp/pull/2618/commits/7651e60a418fe9aad1a69a65018f9def4f2999e2.